src/components/application_manager/src/resumption/resumption_data_processor.cc

```diff
@@ -215,67 +202,100 @@ void ResumptionDataProcessor::ProcessResponseFromHMI(
                   "application id for correlation id "
                       << corr_id << " and function id: " << function_id
                       << " was not found.");
-    request_app_ids_lock_.Release();
-    return;
+    return utils::Optional<uint32_t>::OptionalEmpty::EMPTY;
   }
+  return utils::Optional<uint32_t>(app_id_ptr->second);
+}
 
-  const uint32_t app_id = app_id_ptr->second;
-  request_app_ids_lock_.Release();
-
-  LOG4CXX_DEBUG(logger_, "app_id is: " << app_id);
-
-  LOG4CXX_DEBUG(logger_,
-                "Found function id: " << app_id_ptr->first.function_id
-                                      << " correlation id: "
-                                      << app_id_ptr->first.correlation_id);
-
-  LOG4CXX_DEBUG(logger_,
-                "Now processing event with function id: "
-                    << request_ids.function_id
-                    << " correlation id: " << request_ids.correlation_id);
+utils::Optional<ResumptionRequest> ResumptionDataProcessor::GetRequest(
+    const uint32_t app_id,
+    const hmi_apis::FunctionID::eType function_id,
+    const int32_t corr_id) {
+  sync_primitives::AutoWriteLock lock(resumption_status_lock_);
+  std::vector<ResumptionRequest>& list_of_sent_requests =
+      resumption_status_[app_id].list_of_sent_requests;
 
-  resumption_status_lock_.AcquireForWriting();
   if (resumption_status_.find(app_id) == resumption_status_.end()) {
     LOG4CXX_ERROR(logger_,
                   "No resumption status info found for app_id: " << app_id);
-    resumption_status_lock_.Release();
-    return;
+    return utils::Optional<ResumptionRequest>::OptionalEmpty::EMPTY;
   }
 
-  ApplicationResumptionStatus& status = resumption_status_[app_id];
-  std::vector<ResumptionRequest>& list_of_sent_requests =
-      status.list_of_sent_requests;
-
-  auto request_ptr =
+  auto request_iter =
       std::find_if(list_of_sent_requests.begin(),
                    list_of_sent_requests.end(),
                    [function_id, corr_id](const ResumptionRequest& request) {
                      return request.request_ids.correlation_id == corr_id &&
                             request.request_ids.function_id == function_id;
                    });
 
-  if (list_of_sent_requests.end() == request_ptr) {
+  if (list_of_sent_requests.end() == request_iter) {
     LOG4CXX_ERROR(logger_, "Request not found");
-    resumption_status_lock_.Release();
+    return utils::Optional<ResumptionRequest>::OptionalEmpty::EMPTY;
+  }
+  return utils::Optional<ResumptionRequest>(*request_iter);
+}
+
+utils::Optional<ResumeCtrl::ResumptionCallBack>
+ResumptionDataProcessor::GetResumptionCallback(const uint32_t app_id) {
```

Comment:
Put auto traces in new functions

Reply:
Done in 90d5b8f7a4d8ee35f2581eff4cb63a2126a04f08